Src/IronPython/Runtime/Operations/StringOps.cs

```diff
@@ -1725,11 +1725,103 @@ public static class StringOps {
 
 #if FEATURE_ENCODING
         private static DecoderFallback ReplacementFallback = new DecoderReplacementFallback("\ufffd");
+        
+        private class SurrogateEscapeDecoderFallback : DecoderFallback {
+            private class SurrogateEscapeDecoderFallbackBuffer : DecoderFallbackBuffer {
+                private SurrogateEscapeDecoderFallback _fallback;
+                private int _count = -1;
+                private int _index = -1;
+                private char[] _output = new char[4];
```

Comment:
Maybe I just don't understand how these work. Why is our buffer only 4? Will need to look into how these work some more.